source/blog/2015-06-18-ember-data-1-13-released.md

```diff
@@ -0,0 +1,578 @@
+ready to use the new Serializer API, add `isNewSerializerAPI: true`
+when extending `JSONSerializer` or `RESTSerializer`. The new
+`JSONAPISerializer` uses the new Serializer API by default. The
+`isNewSerializerAPI` flag is **only** required for Ember Data 1.13 and
+will be removed in Ember Data 2.0. The new `JSONAPISerializer` only
+supports the new Serializer API.
+
+Thanks to [@wecc](https://github.com/wecc) for his hard work on
+implementing this feature in a backwards compatible way.
+
+We would also like thank [@kurko](https://github.com/kurko)
+for his efforts maintaining the
+[community JSON-API adapter](https://github.com/kurko/ember-json-api).
+
+### New Errors API
+
+Error handling is an important part of any ambitious
+application. Ember-Data had a support for what we called invalid
+errors from the very beginning. Invalid errors are a type of errors
+that can be corrected by the user, means a "correct and retry" cycle
+is encouraged for this type of errors. But there a other type of
+errors that can occur in the adapter. A network can be down or you can
+reach the quota on your `localStorage`. We wanted to provide some
+better semantics to adapter authors on how to handle this type of
+errors and expos an api to application authors to react on them.
+
+In Ember-Data 1.13 we introduced a new adapter hook:
+`handleResponse`. At the same time we deprecating `ajaxSuccess` and
+`ajaxError`.
+
+There been quite a bit of discussion on what an error in the context
+of an adapter actually mean and how much control should we allow to
+the application on deciding given a particular response. At some point
+we made changes to support returning `InvalidError` from
+`ajaxSuccess`. This is why today we replacing both hooks
+(`ajaxSuccess` and `ajaxError`) with just one (`handleResponse`). The
+default implementation will use a couple of new hooks (`isSuccess` and
+`isInvalid`) to decide if a response is a success or an error of
+certain type. It allows for more control over how your application
+handles errors. Default implementation of `isInvalid` is to return
+true if http status code is 422, but you may desside on some other
+semantics, for example check `code` property on JSON-API error object.
+
+We also wanted for quite some time to abstract `jQuery.ajax`
+semantics. We used to reject from the adapter with `jqXHR` object and
+leave the application to deal with it. This strategy have a few
+drawbacks. It made life hard for application authors by making switch
+from one adapter to another a lot of work. It also made life of
+adapter authors hard by failing to provide them a common set of error
+types that can be shared across different backends. A `TimeoutError`
+for example can express a common type of error across a wide range of
+backends. New hooks take three arguments: status code, a hash of
+response headers and parsed payload. It makes them agnostic about
+underlying implementation and will allow to easily use `fetch` in the
+future.
```

Comment:
@mixonic because in the old implementation we were returning jqXHR object as a response object implicitly coupling adapter and even application code to `jQuery.ajax`.